src/odemis/acq/milling/patterns.py

```diff
@@ -0,0 +1,248 @@
+
+
+class MicroexpansionPatternParameters(MillingPatternParameters):
+    """Represents microexpansion pattern parameters"""
+
+    def __init__(self, width: float, height: float, depth: float, spacing: float, center = (0, 0), name: str = "Trench"):
+        self.name = model.StringVA(name)
+        self.width = model.FloatContinuous(width, unit="m", range=(1e-9, 900e-6))
+        self.height = model.FloatContinuous(height, unit="m", range=(1e-9, 900e-6))
+        self.depth = model.FloatContinuous(depth, unit="m", range=(1e-9, 100e-6))
+        self.spacing = model.FloatContinuous(spacing, unit="m", range=(1e-9, 900e-6))
+        self.center = model.TupleContinuous(center, unit="m", range=((-1e3, -1e3), (1e3, 1e3)), cls=(int, float))
+
+    def to_dict(self) -> dict:
+        """Convert the parameters to a json object"""
+        return {"name": self.name.value,
+                "width": self.width.value,
+                "height": self.height.value,
+                "depth": self.depth.value,
+                "spacing": self.spacing.value,
+                "center_x": self.center.value[0],
+                "center_y": self.center.value[1],
+                "pattern": "microexpansion"
+        }
+
+    @staticmethod
+    def from_dict(data: dict) -> 'MicroexpansionPatternParameters':
+        """Create a MicroexpansionPatternParameters object from a json object"""
+        return MicroexpansionPatternParameters(
+                        width=data["width"],
+                        height=data["height"],
+                        depth=data["depth"],
+                        spacing=data["spacing"],
+                        center=(data.get("center_x", 0), data.get("center_y", 0)),
+                        name=data.get("name", "Microexpansion"))
+
+    def __repr__(self) -> str:
+        return f"{self.to_dict()}"
+
+    def generate(self) -> List[MillingPatternParameters]:
+        """Generate a list of milling shapes for the microscope"""
+        name = self.name.value
+        width = self.width.value
+        height = self.height.value
+        depth = self.depth.value
+        spacing = self.spacing.value / 2
+        center_x, center_y = self.center.value
+
+        patterns = [
+            RectanglePatternParameters(
+                name=f"{name} (Left)",
+                width=width,
+                height=height,
+                depth=depth,
+                rotation=0,
+                center = (center_x - spacing, center_y),
+                scan_direction="TopToBottom",
+            ),
+            RectanglePatternParameters(
+                name=f"{name} (Right)",
+                width=width,
+                height=height,
+                depth=depth,
+                rotation=0,
+                center = (center_x + spacing, center_y),
+                scan_direction="TopToBottom",
+            ),
+        ]
+
+        return patterns
+
+# dictionary to map pattern names to pattern classes
+pattern_generator = {
```

Comment:
That's bit confusing name, as "generator" has a very specific meaning in Python (which is not meant here). Also, let's use uppercase, as it's a constant. Maybe something like `PATTERN_NAME_TO_CLASS`?